File: backend/app/schemas/job_search.py
from typing import Literal

from pydantic import BaseModel, Field


class JobSearchRequest(BaseModel):
    cv_id: int
    query: str | None = None
    location: str = "Remote"
    remote: str | None = None
    jobage: int = 30
    limit: int = Field(default=12, ge=1, le=20)
    level: str | None = None


class JobSearchHit(BaseModel):
    id: str
    title: str
    company: str | None = None
    location: str | None = None
    date: str | None = None
    url: str
    matched_keywords: list[str] = []
    seniority: str | None = None
    category: str | None = None
    source: Literal["linkedin", "freehire"]


class JobSearchResponse(BaseModel):
    query: str
    location: str
    results: list[JobSearchHit]
    note: str
    derived_by: Literal["ai", "deterministic"] = "deterministic"
    derived_level: str | None = None

File: src/types/jobSearch.ts
export interface JobSearchHit {
  id: string
  title: string
  company: string | null
  location: string | null
  date: string | null
  url: string
  matchedKeywords: string[]
  seniority: string | null
  category: string | null
  source: "linkedin" | "freehire"
}

export interface JobSearchResult {
  query: string
  location: string
  results: JobSearchHit[]
  note: string
  derivedBy: "ai" | "deterministic"
  derivedLevel: string | null
}

export interface JobSearchParams {
  cvId: number
  query?: string
  location?: string
  remote?: "remote" | "hybrid" | "onsite"
  jobage?: number
  limit?: number
  level?: string
}
